### PR TITLE
fix dynamic lighting with custom projections

### PR DESCRIPTION
**Pull request by pixelflinger**
Forxelization made many assumptions about the shape of the projection
matrix, the gist of these fixes is to remove those assumptions.

Fix https://github.com/google/filament/issues/6390

### CODE REVIEW

File: filament/src/Froxelizer.cpp [hunk 1]
```diff
@@ -793,40 +760,46 @@ void Froxelizer::froxelizePointAndSpotLight(
...
-    const size_t x0 = imin.first;
-    const size_t y0 = imin.second;
+    const auto [x0, y0] = clipToIndices(pmin);
     const size_t z0 = findSliceZ(znear);
 
-    const auto imax = clipToIndices(max(xyRightNear, xyRightFar));
-    const size_t x1 = imax.first  + 1;  // x1 points to 1 past the last value (like end() does
-    const size_t y1 = imax.second;      // y1 points to the last value
-    const size_t z1 = findSliceZ(zfar); // z1 points to the last value
+    const auto [x1, y1] = clipToIndices(pmax);
```
Comment (romainguy) [inline filament/src/Froxelizer.cpp:799]: x1 was `imax.first+1` and now it's just `imax.first`, is that expected?

File: filament/src/Froxelizer.cpp [hunk 2]
```diff
@@ -838,43 +811,41 @@ void Froxelizer::froxelizePointAndSpotLight(
...
-                    // The reduced sphere from the previous stage intersects this horizontal plane
+                    // The reduced sphere from the previous stage intersects this horizontal plane,
                     // and we now have new smaller sphere centered on these two previous planes
                     size_t bx = std::numeric_limits<size_t>::max(); // horizontal begin index
                     size_t ex = 0; // horizontal end index
 
-                    // find the begin index (left side)
-                    for (size_t ix = x0; ix < x1; ++ix) {
+                    // find the "begin" index (left side)
+                    for (size_t ix = x0; ix < x1 + 1; ++ix) {
```
Comment (romainguy) [inline filament/src/Froxelizer.cpp:843]: I assume that's where the + 1 now is right?

**Review by romainguy — APPROVED** [review]

File: filament/src/Froxelizer.cpp (hunk 2 above)
Comment (pixelflinger) [reply filament/src/Froxelizer.cpp:843]: yes

**Comment by kpeeters** [comment]
The fix in ma/froxelizer does the trick, thanks a lot!

**Pull request merged**